src/common/config.go

```diff
@@ -305,6 +355,64 @@ func SandboxConfJson() string {
 	return string(s)
 }
 
+// SaveConfigAtomic saves a config to a file atomically to prevent race conditions
+// This is used for template.json creation to avoid corruption during concurrent access
```

Comment:
Do we actually have concurrent access, or is the real concern that the data is not flushed to disk before the disk snapshot begins?

Is there any downside to just having a regular SaveConfig that always takes special care with sync/rename?